platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/turtle/runner/Turtle.java

```diff
@@ -222,4 +225,21 @@ private void tickAllNodes() {
             }
         }
     }
+
+    public void validate() {
+        for (final TurtleNode node : nodes) {
+            List<ReservedSignedState> reservedSignedStates =
+                    node.getSignedStateHolder().getCollectedSignedStates();
+
+            for (final ReservedSignedState reservedSignedState : reservedSignedStates) {
+                assertTrue(reservedSignedState.get().isComplete());
+                assertTrue(reservedSignedState.get().isVerifiable());
+                assertTrue(reservedSignedState.get().getCreationTimestamp().isBefore(Instant.now()));
+                assertTrue(reservedSignedState
```

Comment:
We can additionally test if sigSet contains a node which is not part of the current roster. This would mean a wrong signature was collected by the StateSignatureCollectors, which will show us a bug. Currently this is unlikely to fail, but with the integration of the dynamic rosters this would be a key validation check.